specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-04-01/vaults.json

```diff
@@ -0,0 +1,1803 @@
+      "description": "The details of the identity used for CMK",
+      "properties": {
+        "useSystemAssignedIdentity": {
+          "type": "boolean",
+          "description": "Indicate that system assigned identity should be used. Mutually exclusive with 'userAssignedIdentity' field"
+        },
+        "userAssignedIdentity": {
+          "type": "string",
+          "description": "The user assigned identity to be used to grant permissions in case the type of identity used is UserAssigned"
+        }
+      }
+    },
+    "OperationResource": {
+      "type": "object",
+      "description": "Operation Resource",
+      "properties": {
+        "endTime": {
+          "description": "End time of the operation",
+          "format": "date-time",
+          "type": "string"
+        },
+        "error": {
+          "$ref": "#/definitions/Error",
+          "description": "Required if status == failed or status == canceled. This is the OData v4 error format, used by the RPC and will go into the v2.2 Azure REST API guidelines."
+        },
+        "id": {
+          "description": "It should match what is used to GET the operation result",
+          "type": "string"
+        },
+        "name": {
+          "description": "It must match the last segment of the \"id\" field, and will typically be a GUID / system generated value",
+          "type": "string"
+        },
+        "status": {
+          "description": "The status of the operation. (InProgress/Success/Failed/Cancelled)",
+          "type": "string"
+        },
+        "startTime": {
+          "description": "Start time of the operation",
+          "format": "date-time",
+          "type": "string"
+        }
+      }
+    },
+    "CloudError": {
+      "description": "An error response from Azure Backup.",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/Error"
+        }
+      },
+      "title": "CloudError",
+      "x-ms-external": true
+    },
+    "Error": {
+      "description": "The resource management error response.",
+      "properties": {
+        "additionalInfo": {
+          "description": "The error additional info.",
+          "items": {
+            "$ref": "#/definitions/ErrorAdditionalInfo"
+          },
+          "readOnly": true,
+          "type": "array"
+        },
+        "code": {
+          "description": "The error code.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "details": {
+          "description": "The error details.",
+          "items": {
+            "$ref": "#/definitions/Error"
+          },
+          "readOnly": true,
+          "type": "array"
+        },
+        "message": {
+          "description": "The error message.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "target": {
+          "description": "The error target.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "innerError": {
```

Comment:
>innerError [](http://example.com/codeflow?start=9&length=10)

details is usually used to write inner error details, how is this 'innerError' different than 'details' property?

Reply:
Details -> An array of JSON objects that MUST contain name/value pairs for code and message, and MAY contain a name/value pair for target

Inner Error -> The contents of this object are service-defined. Usually this object contains information that will help debug the service. This is generally used in the development environments and not in the production environments. We send null in the production environment

Reply:
The problem I see here is that you are making a property to be environment dependent, this is usually modeled as a discriminator with an enum (e.g. environmentType, depending on this type value you can show different information to the user and the user is well aware of it).

If this is for Dev environment, and since this seems to be a single string value, consider modeling this as part of your details property, you can create an error code that identifies that this is a Dev error, this way users can easily identify your error type, otherwise you'll making users to possibly determine their environment based on innerError having a value or not, which is not ideal.

Reply:
I will be removing the inner error section and then will discuss among my team how to fix the code.

Reply:
Updated